synse_server/api/websocket.py

```diff
@@ -312,21 +345,17 @@ def error(msg_id=None, message=None, ex=None) -> str:
 
     logger.info(_('new websocket connection'), source=request.ip)
 
-    while True:
-        r = await ws.recv()
+    async for message in ws:
```

Comment:
thinking through the process for cancelling a stream without closing the websocket, we could send a message to signal an end to the stream. the issue there is that the streaming response handling effectively blocks this loop, so any future message we receive while streaming will be queued up, but not processed. 

likely, the 'easiest' approach to this would be to run the streamed reading collection in a separate thread. similar to how the threaded stream handlers are set up now for individual plugins, this thread can be tracked and canceled. With this, this loop should not be blocked from processing future messages, and the graceful cancellation of the collector thread should also be able to trigger the graceful cancellation of the plugin streamer threads.

I'm now leaning towards resolving this in this PR, as having to close a websocket to stop the stream, and having the websocket be blocked on stream is not an ideal usage.

This will require a bit more work and probably some refactoring of the websocket handler here, so it'll be a day or two until this is ready for re-review.